app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepReducer.js

```diff
@@ -17,30 +17,69 @@ const initialState = Immutable({
   conflict_vms: []
 });
 
-const _formatValidVms = vms =>
-  vms &&
-  vms.map(v => {
-    v.valid = true;
-    v.allocated_size = numeral(v.allocated_size).format('0.00b');
-    v.reason = V2V_VM_POST_VALIDATION_REASONS[v.reason];
-    return v;
-  });
-const _formatInvalidVms = vms =>
-  vms &&
-  vms.map(v => {
-    v.allocated_size = numeral(v.allocated_size).format('0.00b');
-    v.reason = V2V_VM_POST_VALIDATION_REASONS[v.reason];
-    if (
-      v.reason === V2V_VM_POST_VALIDATION_REASONS.migrated ||
-      v.reason === V2V_VM_POST_VALIDATION_REASONS.in_other_plan
-    ) {
-      v.warning = true;
-    } else {
-      v.invalid = true;
-    }
+const manageDuplicateVMRows = (vm, vmIndex, uniqueIds) => {
+  const index = vm.id && uniqueIds.indexOf(vm.id);
+  if (index > -1) {
+    uniqueIds.splice(index, 1);
+  } else if (index === -1) {
+    vm.reason = V2V_VM_POST_VALIDATION_REASONS.duplicate;
+    vm.warning = false;
+    vm.valid = false;
+    vm.invalid = true;
+    vm.id = `duplicate-${vm.id}-${vmIndex}`;
+  }
+};
+
+const manageBlankReason = vm => {
+  if (!vm.reason) {
+    vm.reason = V2V_VM_POST_VALIDATION_REASONS.no_info_available;
+  }
+};
+
+const manageOddCSVImportErrors = (vm, vmIndex, uniqueIds) => {
+  manageDuplicateVMRows(vm, vmIndex, uniqueIds);
+  manageBlankReason(vm);
+};
+
+const _formatValidVms = vms => {
+  const uniqueIds = vms && [...new Set(vms.map(value => value.id))];
+  let vIndex = 0;
+  return (
+    vms &&
+    vms.map(v => {
+      vIndex += 1;
```

Comment:
no need to keep a counter here if on line 49 yah do something like `vms.map((v, index)  => {`

then on line 54 `index + 1` will give you that plus one offset

Reply:
Thanks! That's a good tip :)

Reply:
9774ad0